src/docs/arch-eviction.dox

```diff
@@ -1,15 +1,80 @@
+extracts all the pages that can be evicted, sorts them, and takes 1/3rd of the evictable
+candidates to do an approximate LRU and pushes them into the eviction_queues. The whole
+goal is to remove the content which is older to be evicted and put it in the queue. A user can
+configure a minimum and a maximum number of eviction worker threads to be created through
+the connection configuration string. The goal of eviction worker threads is to pop the
+pages from the queues and evict them. Eviction worker threads wait for queues to be populated,
+working with the Server through a CV signaling mechanism. They then get to work
+independently with the server, till the queues contain enough work for the candidates. There is
+a mechanism where the number of worker threads can be scaled dynamically between the range
+depending upon if we are getting more performance with more workers or getting still good
+performance and do context switching.
+
+It is possible to have just the eviction server and no worker threads. In such a case,
+the eviction server walks the tree to find pages to evict and evict these pages from the
+cache. Pages that are being marked for forced eviction go onto the urgent
+queue, and these pages take precedence over the pages from the ordinary eviction queues.
+
+If there are other threads reading the content, the page become un-evictable, and the
+Eviction server/worker threads check for that. Eviction has to lock and get exclusive
+access to the page so that after having checked if the page is evictable and starting to
+evict, a parallel reader doesn't start to access it.
+
+@section clean_dirty_data Clean vs dirty data
+- Clean refers to data in the cache that is identical to the version stored on disk.
+- Dirty refers to data in the cache that has been modified and needs to be reconciled
+  with the data stored in disk.
+
+@section clean_dirty_eviction Clean vs dirty eviction
+- In the case of clean eviction, there is no dirty content on the page and it is directly
+  removed from the memory. After that, the page remains solely on the disk unchanged.
+- In the case of dirty eviction, the dirty page goes through reconciliation. The obsolete
+  content on the page is thrown out, the latest value goes to the data store and older
+  values go to the history store.
+
+Eviction triggers
+
+WiredTiger provides several configuration options for tuning how pages are evicted from
+the cache.
+
+WiredTiger eviction tuning options can be configured when first opening
+a database via ::wiredtiger_open, or changed after open with
+WT_CONNECTION::reconfigure.
+
+The \c eviction_target configuration value is the level at
+which WiredTiger attempts to keep the overall cache usage.  Eviction worker
+threads are active when the cache contains at least this much content,
+expressed as a percentage of the total cache size.
+
+The \c eviction_trigger configuration value is the level at which application threads
+start to perform the eviction.
```

Comment:
We should add a sentence about throttling the workload when eviction is unable to keep up with the workload in maintaining the cache in the desired state. In such a case application threads are also forced to do the work of eviction workers threads, before they can read or write more data. Ideally we should talk about this before "Eviction triggers and targets".

Reply:
Fixed